### ISSUE TITLE
persist: connection pool PR causes table SELECTs to hang

### ISSUE THREAD

**Issue by danhhz** [MaterializeInc/materialize#13858]
This hangs on main

```
materialize=> create table t();
CREATE TABLE
materialize=> select * from t;
```

It repros at https://github.com/MaterializeInc/materialize/pull/13794 and does not repro at the commit before that PR. We're reverting that PR while we investigate.

**Comment by danhhz** [comment]
It's curious that there seems to be no logs from persist. I took a quick peek at the raw metrics (no dashboard hooked up yet) but nothing jumped out at me.

Attempting to figure out how this passed CI: tried setting consensus_connection_pool_max_size to 1, no effect. Then ran `cargo run --bin testdrive -- 'test/testdrive/tables.td'` which seems to immediately hang. This likely points to some sort of merge skew.

**Comment by danhhz** [comment]
This was closed by #13897
